source/routing/raptor_api.cpp

```diff
@@ -51,6 +51,126 @@ namespace navitia { namespace routing {
 // Usefull to protect raptor loop
 static const uint max_nb_raptor_call = 100;
 
+/**
+ * @brief This function determine the break condition that depends on nb of found journeys,
+ * min_nb_journeys and time frame.
+ *
+ * @details
+ *
+ *   T           T + N          T + 24H
+ *   -----------------------------------
+ *
+ * timeframe_duration (T + N): the end datetime of original request's timeframe
+ * max_duration (T + 24H): the max datetime compared to the orignal datetime,
+ *                                   this param is used in case when T + N is reached but
+ *                                   we still don't have enough journeys(total_nb_journeys<min_nb_journeys),
+ *                                   we allow raptor to get out of the requested time frame (T + N) but still
+ *                                   limit the search frame inside bound
+ *
+ *                   min_nb_journeys is set?
+ *                    /               \
+ *                   / Y               \ N
+ *          time frame is set?        time frame is set?
+ *            /       \                  /     \
+ *           / N       \                /N      \Y
+ *       Case 1         \            Case 4     Case 5
+ *                       \ Y
+ *      total_nb_journeys < min_nb_journeys?
+ *                   /        \
+ *                  /Y         \N
+ *              Case 2       Case 3
+ *
+ *
+ *
+ */
+static bool keep_going(const uint32_t total_nb_journeys,
+                       const uint32_t nb_try,
+                       const bool clockwise,
+                       const DateTime request_date_secs,
+                       const boost::optional<uint32_t>& min_nb_journeys,
+                       const boost::optional<uint32_t>& timeframe_duration) {
```

Comment:
OK, this is super confusing (and this is the case quite some times in this PR): This is a datetime, no? Absolutely not a duration.
It's the same type underneath, but at least we need to name it correctly (we should definitely type it).